### PR TITLE
feat: Bump dependencies to support new lockfile version

### PR DESCRIPTION
**Pull request by pavelzw**
# Motivation

pixi 0.39.0 has a new lockfile version that includes Conda Source packages
closes #78

# Changes

Update rattler and adjust tests accordingly

### CODE REVIEW

File: src/pack.rs [hunk 1]
```diff
@@ -235,25 +233,27 @@ fn reqwest_client_from_auth_storage(auth_file: Option<PathBuf>) -> Result<Client
...
-        .ok_or(anyhow!("could not get file name"))?;
+    let file_name = &package.file_name;
     let mut dest = File::create(output_dir.join(file_name)).await?;
 
-    tracing::debug!("Fetching package {}", package.url());
-    let mut response = client.get(package.url().to_string()).send().await?;
+    tracing::debug!("Fetching package {}", package.location);
+    let url = match &package.location {
+        UrlOrPath::Url(url) => url,
+        UrlOrPath::Path(path) => anyhow::bail!("Path not supported: {}", path),
```
Comment (pavelzw) [inline src/pack.rs:250]: probably related to #79

File: src/pack.rs [hunk 2]
```diff
@@ -78,12 +78,13 @@ pub async fn pack(options: PackOptions) -> Result<()> {
...
 
-    let mut conda_packages_from_lockfile: Vec<CondaPackage> = Vec::new();
+    let mut conda_packages_from_lockfile: Vec<CondaBinaryData> = Vec::new();
 
     for package in packages {
         match package {
-            Package::Conda(p) => conda_packages_from_lockfile.push(p),
-            Package::Pypi(_) => {
+            LockedPackageRef::Conda(CondaPackageData::Binary(binary_data)) => conda_packages_from_lockfile.push(binary_data.clone()),
+            LockedPackageRef::Conda(CondaPackageData::Source(_)) => anyhow::bail!("Conda source packages are not yet supported by pixi-pack"),
```
Comment (pavelzw) [inline src/pack.rs:86]: xref #80

**Comment by pavelzw** [comment]
```
    test_line_endings::case_1
    test_line_endings::case_2
    test_reproducible_shasum::case_1
    test_reproducible_shasum::case_2
    test_reproducible_shasum::case_4
    test_reproducible_shasum::case_5
    test_reproducible_shasum::case_6
    test_run_packed_executable
```

are failing because it tries to download pixi-pack 0.3.1 from github which hasn't released yet

File: src/pack.rs (hunk 1 above)
Comment (delsner) [reply src/pack.rs:254]: Will this be addressed by #71?

**Review by delsner — APPROVED** [review]
LGTM! Thanks.

File: src/pack.rs (hunk 1 above)
Comment (pavelzw) [reply src/pack.rs:254]: yes, meant #71

**Pull request merged**